### PR TITLE
Updated broken links

### PR DESCRIPTION
**Pull request by dreesefadil**
## Product Description

A number of Confluence links were broken after transitioning to Confluence Cloud. This PR replaces most broken Confluence links with the correct links. There are a few links that are either internal or no longer relevant, so those links were removed.

I also ran into a few linting issues, so I fixed those.

## Technical Summary

https://dimagi.atlassian.net/browse/SAAS-15729

The broken links were found using the find_broken_links script in here: https://github.com/dimagi/commcare-hq/pull/34877

Kai then added in the correct links:
https://docs.google.com/spreadsheets/d/1e7QSxtTiFC4swM1oHto81J-CXNHxn2Bvtz3jzvKNDps/edit?gid=823348706#gid=823348706

### Safety story

I double checked that all of the new links worked before adding them.

### QA Plan

N/A

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by mkangia** [comment]
Hey @dreesefadil 

Thanks for these.
Those are a lot of changes, so might take time for someone to review.

Its usually easier for reviewers if the PR has limited changes. So, keeping this limited to just the links would have made it quicker and then lint fixes in a separate PR. Definitely not discouraging making lint fixes and other improvements to the code.

**Comment by mkangia** [comment]
No worries @dreesefadil

File: corehq/apps/app_manager/templates/app_manager/partials/releases/releases_deploy_modal.html
Comment (nospame) [inline corehq/apps/app_manager/templates/app_manager/partials/releases/releases_deploy_modal.html:170]: nit: double quotes in html

```suggestion
                                <a href="https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2143946239/Installing+CommCare+on+Android+Devices">
```

File: corehq/apps/export/templates/export/customize_export_new.html
Comment (nospame) [inline corehq/apps/export/templates/export/customize_export_new.html:177]: nit: Seems like we're including friendly names in links, in which case this would be:
```suggestion
                                       for the <a href="https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2143955952/CommCare+Data+Export+Tool+DET" target="_blank">
```

File: corehq/ex-submodules/soil/templates/soil/partials/dl_status.html
Comment (nospame) [inline corehq/ex-submodules/soil/templates/soil/partials/dl_status.html:23]: The updated/broken links sheet calls this feature dead. Should this whole `{% if allow_dropbox_sync %}` section be removed? Maybe a question for @kcowger?

File: corehq/feature_previews.py
Comment (nospame) [inline corehq/feature_previews.py:162]: This link needs to be updated or removed too, right? The page it redirects to is archived, so my bet is "removed".

File: corehq/messaging/scheduling/views.py
Comment (nospame) [inline corehq/messaging/scheduling/views.py:1073]: This wasn't you, but ideally we should specify which checks flake8 is skipping:
```suggestion
                "help_link": "https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2143956488/Conditional+Alerts",  # noqa: E501
```

File: corehq/toggles/__init__.py
Comment (nospame) [inline corehq/toggles/__init__.py:2861]: This one should be fine to keep because it links to (in progress) documentation for the feature flag it is the `help_link` on.

**Review by nospame — COMMENTED** [review]
Overall looks good - a few non-blocking comments plus a couple it'd be good to get more info on before approval.

File: corehq/apps/app_manager/static/app_manager/json/commcare-profile-settings.yml
Comment (mjriley) [inline corehq/apps/app_manager/static/app_manager/json/commcare-profile-settings.yml:249]: This looks like it could be more specific. The previous link pointed to resizing images, and I see that there's a section within the new link that would probably be nicer: https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2143946731/Images+in+CommCare#Auto-Resize-Images-on-Android

File: corehq/apps/app_manager/static_strings.py
Comment (mjriley) [inline corehq/apps/app_manager/static_strings.py:49]: Same link -- could probably make this more specific

File: corehq/apps/app_manager/static_strings.py
Comment (mjriley) [inline corehq/apps/app_manager/static_strings.py:206]: This also appears like it could be more specific: https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2143946731/Images+in+CommCare#CommCare-Developer-Toolkit-Image-Sizing-with-Multiple-Android-Device-Models

File: corehq/apps/app_manager/templates/app_manager/app_exchange.html
Comment (mjriley) [inline corehq/apps/app_manager/templates/app_manager/app_exchange.html:31]: Did we get rid of this page?

File: corehq/apps/app_manager/templates/app_manager/partials/releases/releases_table.html
Comment (mjriley) [inline corehq/apps/app_manager/templates/app_manager/partials/releases/releases_table.html:78]: Looks like there's a more specific section on deprecating? https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2143944977/Data+Dictionary#Deprecating-%26-Delete-Case-Types-and-Case-Properties

File: corehq/apps/app_manager/templates/app_manager/partials/settings/supported_languages.html
Comment (mjriley) [inline corehq/apps/app_manager/templates/app_manager/partials/settings/supported_languages.html:44]: This points to a table of contents page that I think the user wouldn't be happy about. Is this link what was previously linked to? https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2143958174/Language+Configuration+in+CommCareHQ

File: corehq/apps/case_importer/templates/case_importer/excel_config.html
Comment (mjriley) [inline corehq/apps/case_importer/templates/case_importer/excel_config.html:23]: More specific: https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2143946828/Importing+Cases+Using+Excel#Update-Multiple-Case-Types-in-a-Single-Import

File: corehq/apps/data_dictionary/templates/data_dictionary/partials/confirmation_modals.html
Comment (mjriley) [inline corehq/apps/data_dictionary/templates/data_dictionary/partials/confirmation_modals.html:42]: More specific: https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2143944977/Data+Dictionary#Deprecating-%26-Delete-Case-Types-and-Case-Properties

File: corehq/apps/data_interfaces/templates/data_interfaces/explore_case_data.html
Comment (mjriley) [inline corehq/apps/data_interfaces/templates/data_interfaces/explore_case_data.html:79]: Just want to make sure this is the right link? I wouldn't be surprised if we changed 'feature preview' to 'add-on', but it's awkward that the translated text above still says "This is a Feature Preview". If we've changed the term, I'd imagine we'd also want to update the translation block.

File: corehq/apps/export/templates/export/partials/table.html
Comment (mjriley) [inline corehq/apps/export/templates/export/partials/table.html:356]: I think https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2143956271/Form+Data+Export#Multimedia-Exports.1 is the correct link here? It specifically references the file size limitation

File: corehq/apps/reports/filters/users.py
Comment (mjriley) [inline corehq/apps/reports/filters/users.py:205]: More specific: https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2215051298/Organization+Data+Management#Search-for-Locations

File: corehq/apps/reports/templates/reports/reportdata/bootstrap3/case_data.html
Comment (mjriley) [inline corehq/apps/reports/templates/reports/reportdata/bootstrap3/case_data.html:39]: More specific: https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2143944977/Data+Dictionary#Deprecating-%26-Delete-Case-Types-and-Case-Properties

File: corehq/apps/reports/templates/reports/reportdata/bootstrap5/case_data.html
Comment (mjriley) [inline corehq/apps/reports/templates/reports/reportdata/bootstrap5/case_data.html:39]: More specific: https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2143944977/Data+Dictionary#Deprecating-%26-Delete-Case-Types-and-Case-Properties

File: corehq/ex-submodules/soil/templates/soil/partials/dl_status.html
Comment (mjriley) [inline corehq/ex-submodules/soil/templates/soil/partials/dl_status.html:19]: Confirming that these links were removed?

File: corehq/feature_previews.py
Comment (mjriley) [inline corehq/feature_previews.py:103]: More specific: https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2143951603/Case+List+and+Case+Detail+Configuration#Calculations-in-the-Case-List%2FDetail

File: corehq/feature_previews.py
Comment (mjriley) [inline corehq/feature_previews.py:160]: Did we just remove the documentation on call centers, but leave in the preview?

File: corehq/messaging/scheduling/views.py
Comment (mjriley) [inline corehq/messaging/scheduling/views.py:1073]: More specific: https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2143956488/Conditional+Alerts#Bulk-download-and-upload-of-SMS-content-in-conditional-alerts

File: docs/fhir/index.rst
Comment (mjriley) [inline docs/fhir/index.rst:103]: More specific: https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2279637003/CommCare+API+Overview#API-Authentication

File: docs/web_apps.rst
Comment (mjriley) [inline docs/web_apps.rst:109]: More specific: https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2143951603/Case+List+and+Case+Detail+Configuration#Minimize-Duplicates%3A-Registration-From-the-Case-List

**Review by mjriley — APPROVED** [review]
Looks good -- all my feedback is around more specific links we could provide, but as the links in this PR all work and were sourced from the spreadsheet, I'm not going to block this for the updates. I do think our customers would appreciate more specific links, however

File: corehq/apps/export/templates/export/partials/table.html
Comment (kishansampat) [reply corehq/apps/export/templates/export/partials/table.html:356]: @dreesefadil @mjriley the correct link here is https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2337669143/Setting+up+a+Data+Pipeline+via+CommCare

That error message pops up when the amount of data in a UI based export has exceeded 100,000 rows. The one that Matt pointed to is about the size of the multimedia export, which is a separate error message.

**Pull request merged**
